hazelcast-client-legacy/src/main/java/com/hazelcast/client/cache/impl/AbstractClientCacheProxyBase.java

```diff
@@ -201,6 +201,10 @@ public void run() {
                 if (completionListener != null) {
                     completionListener.onException(e);
                 }
+            } catch (Throwable t) {
```

Comment:
It looks like it will work. I am thinking about if NativeOutOfMemoryError should have been Exception not Error in the first place. Other than that, pr looks good.